webhook_to_fedora_messaging_messages/forgejo/utils.py

```diff
@@ -0,0 +1,134 @@
+    """
+    Obtain text specification based on event type
+    """
+    if event_type == "push":
+        return _summarize_push_event(data)
+    elif event_type == "fork":
+        return _summarize_fork_event(data)
+    elif event_type == "pull_request":
+        return _summarize_pull_request_event(data)
+    elif event_type == "issues":
+        return _summarize_issues_event(data)
+    elif event_type == "issue_comment":
+        return _summarize_issue_comment_event(data)
+    else:
+        return "Event type not supported"
+
+
+def _summarize_push_event(data) -> str:
+    """
+    Obtain text specification for push event
+    """
+    repository_name = data["repository"]["full_name"]
+    repository_url = data["repository"]["html_url"]
+    pusher = data["pusher"]["full_name"]
+    pusher_url = data["pusher"]["html_url"]
+    branch = data["ref"].split("/")[-1]
+    commits = "\n".join(
+        [
+            f"Commit: {commit['message']} by {commit['author']['name']} ({commit['url']})"
+            for commit in data["commits"]
+        ]
+    )
+    return (
+        f"Event: Push\n"
+        f"Repository: {repository_name} ({repository_url})\n"
+        f"Pusher: {pusher} ({pusher_url})\n"
+        f"Branch: {branch}\n"
+        f"Commits:\n{commits}"
+    )
+
+
+def _summarize_fork_event(data) -> str:
+    """
+    Obtain text specification for fork event
+    """
+    repository_name = data["repository"]["full_name"]
+    repository_url = data["repository"]["html_url"]
+    forkee = data["forkee"]["full_name"]
+    forkee_url = data["forkee"]["html_url"]
+    owner = data["forkee"]["owner"]["full_name"]
+    owner_url = data["forkee"]["owner"]["html_url"]
+    return (
+        f"Event: Fork\n"
+        f"Repository: {repository_name} ({repository_url})\n"
+        f"Forkee: {forkee} ({forkee_url})\n"
+        f"Owner: {owner} ({owner_url})"
+    )
+
+
+def _summarize_pull_request_event(data) -> str:
+    """
+    Obtain text specification for pull request event
+    """
+    repository_name = data["repository"]["full_name"]
+    repository_url = data["repository"]["html_url"]
+    pr_title = data["pull_request"]["title"]
+    pr_url = data["pull_request"]["html_url"]
+    pr_author = data["pull_request"]["user"]["full_name"]
+    pr_source_branch = data["pull_request"]["head"]["ref"]
+    pr_dest_branch = data["pull_request"]["base"]["ref"]
+    pr_reviewers = ", ".join(
+        [reviewer["full_name"] for reviewer in data["pull_request"]["requested_reviewers"]]
+    )
+    status = data["action"]
+    return (
+        f"Event: Pull Request\n"
+        f"Repository: {repository_name} ({repository_url})\n"
+        f"Title: {pr_title}\n"
+        f"URL: {pr_url}\n"
+        f"Author: {pr_author}\n"
+        f"Status: {status}\n"
+        f"Source Branch: {pr_source_branch}\n"
+        f"Destination Branch: {pr_dest_branch}\n"
+        f"Reviewers: {pr_reviewers}"
```

Comment:
Those are actually the "requested reviewers", not the actual reviewers of the PR, so just having "Reviewers:" can be misleading, I think it should be "Requested Reviewers". And maybe add another field for the assignee ? (who's going to be the actual reviewer of the PR).

Reply:
Aren't assignees supposed to be the folks responsible for merging (eg. fixing merge conflicts if any etc.) while reviewers supposed to be folks reviewing (eg. approving contributions, suggestions changes etc.) the code? This is the same case for the updated GitHub schema as well (which would also require changing hence) should we rename this to "Requested Reviewers".

Reply:
My point is that the requested reviewers aren't necessarily the reviewers. When a PR is sent, the author can ask for the entire team to review it, but that's not who is actually going to review the PR.

Reply:
Fair enough. I will change the key here from `Reviewers` to `Requested Reviewers` in the text return.